pep-0726.rst

```diff
@@ -0,0 +1,171 @@
+PEP: 726
+Title: Module __setattr__ and __delattr__
+Author: Sergey B Kirpichev <skirpichev@gmail.com>
+Sponsor: Adam Turner <python@quite.org.uk>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 24-Aug-2023
+Python-Version: 3.13
+Post-History: `06-Apr-2023 <https://discuss.python.org/t/extend-pep-562-with-setattr-for-modules/25506/>`__
+
+
+Abstract
+========
+
+It is proposed to support ``__setattr__`` and ``__delattr__`` methods
+defined on modules to extend customization of module attribute access
+beyond :pep:`562`.
+
+Motivation
+==========
+
+There are several potential uses of a module ``__setattr__``:
+
+1. To prevent setting attribute at all (make one read-only)
+2. To validate the value to be assigned
+3. To intercept setting an attribute and update some other state
+
+Proper support for read-only attributes would also require adding
+``__delattr__`` helper function to prevent their deletion as well.
+
+Typical workaround is assigning ``__class__`` of a module object to a
+custom subclass of :py:class:`python:types.ModuleType` (see [1]_).
+Unfortunately, this also coming with a noticiable speed regression
```

Comment:
Have you confirmed that this speed regression does NOT happen with the proposed change? If so, I would definitely mention that here (possibly alongside the "would be convenient" on the next line).

Reply:
sp "noticiable" s/be "noticeable"

Reply:
> Have you confirmed that this speed regression does NOT happen with the proposed change?

Yes, that [was discussed in the ideas thread](https://discuss.python.org/t/extend-pep-562-with-setattr-for-modules/25506/10?u=skirpichev).

> I would definitely mention that here

It's mentioned in the "Backwards compatibility" section.  Do you think it's a wrong place?